### PR TITLE
Mother of all parser updates

### PR DESCRIPTION
**Pull request by theHamsta**
- @stsewd @bool3max #948 Still problems with C parser with this change?

- Address #972:
     - pin jsonc parser
     - add tar download support for Gitlab

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -313,7 +313,7 @@ list.vue = {
 
 list.jsonc = {
   install_info = {
-    url = "https://gitlab.com/WhyNotHugo/tree-sitter-jsonc.git",
+    url = "https://gitlab.com/WhyNotHugo/tree-sitter-jsonc",
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:316]: I checked that this is still working with Git

**Comment by theHamsta** [comment]
Maybe it's necessary to remove the nvim version of the C queries from the Neovim repo.

**Review by stsewd — APPROVED** [review]
Yeah, this fixes the issue, but we still need to figure out the `type` and `value` field names giving these errors with ts-cli 0.17.x, not all parsers can remove the field name from the queries.

**Comment by theHamsta** [comment]
I don't think this error really exists and it's just the values just got lost some how for introspection. Are `type` and `value` somehow special in Javascript or in the Lua code that generates that introspection table?

Neovim never had those fields in it's Lua dictionary (since I built the linter) and we had never had a problem with it.

**Comment by stsewd** [comment]
value and field work correctly in the web interface from grammars generated with ts 17.x last time I checked (can double check this again later), so I do think there is something with the neovim integration or with the c api. I'll try to debug this more from the c side of neovim in the following weeks.

**Comment by theHamsta** [comment]
@stsewd but value and field also work correctly in our playground! The Rust CLI or the node process that generate the grammar generate some meta info about the parser that is not actually used for parsing (like node_types.json that has `name` and `type` in their json schema and also some) and also https://github.com/stsewd/tree-sitter-rst/blob/1968158eb612fbfae466584dd438634874e59dbe/src/parser.c#L861-L866.

**Comment by theHamsta** [comment]
The field names that are missing in the introspection table are always the last ones (`name` in RST, `value` in python). So probably an off-by-one errror in `inspect_language`
```

static const char *ts_field_names[] = {
  [0] = NULL,
  [field_alias] = "alias",
  [field_alternative] = "alternative",
  [field_argument] = "argument",
  [field_arguments] = "arguments",
  [field_attribute] = "attribute",
  [field_body] = "body",
  [field_cause] = "cause",
  [field_code] = "code",
  [field_condition] = "condition",
  [field_consequence] = "consequence",
  [field_definition] = "definition",
  [field_function] = "function",
  [field_key] = "key",
  [field_left] = "left",
  [field_module_name] = "module_name",
  [field_name] = "name",
  [field_object] = "object",
  [field_operator] = "operator",
  [field_parameters] = "parameters",
  [field_return_type] = "return_type",
  [field_right] = "right",
  [field_subscript] = "subscript",
  [field_superclasses] = "superclasses",
  [field_type] = "type",
  [field_value] = "value",
};
```

**Comment by stsewd** [comment]
> The field names that are missing in the introspection table are always the last ones (name in RST, value in python). So probably an off-by-one errror in inspect_language

that's a good theory to check.

btw, this is a branch I'm using to test https://github.com/stsewd/tree-sitter-rst/pull/7/ my rst parser generated with ts-cli 0.17.x

**Comment by theHamsta** [comment]
https://github.com/neovim/neovim/blob/c1fbc2ddf15b2f44b615f90b2511349ab974cb83/src/nvim/lua/treesitter.c#L225-L228

The field indices actually start with 1.

**Comment by theHamsta** [comment]
Apparently, yes
```
const char *ts_language_field_name_for_id(
  const TSLanguage *self,
  TSFieldId id
) {
  uint32_t count = ts_language_field_count(self);
  if (count && id <= count) {
    return self->field_names[id];
  } else {
    return NULL;
  }
}
```
will submit a patch.

**Comment by stsewd** [comment]
> Maybe it's necessary to remove the nvim version of the C queries from the Neovim repo.

Yeah, I still get the errors if the runtime path is set explicitly to neovim/runtime with $VIMRUNTIME

**Comment by stsewd** [comment]
BTW,I think we should just merge this, the other problem doesn't happen with other parsers yet (they aren't being generated with ts-cli 0.17.x or don't use all their fields)

**Pull request merged**